solarwinds_apm/configurator.py

```diff
@@ -437,6 +448,100 @@ def _configure_metrics_exporter(
+        self,
+        apm_config: SolarWindsApmConfig,
+    ) -> None:
+        """Configure OTel OTLP logs exporter if enabled.
+        Settings precedence: OTEL_* > SW_APM_EXPORT_LOGS_ENABLED.
+        Links to new global LoggerProvider."""
+        otel_ev = os.environ.get(
+            _OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED
+        )
+        otlp_log_enabled = SolarWindsApmConfig.convert_to_bool(otel_ev)
+        sw_enabled = apm_config.get("export_logs_enabled")
+        if otlp_log_enabled is False:
+            logger.debug(
+                "OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED false. Skipping init of logs exporters"
+            )
+            return
+
+        # If otel_ev is True, ignore sw_enabled.
+        # If otel_ev is None (unset or could not convert to bool)
+        # then sw_enabled determines logs export setup.
+        if not otlp_log_enabled and sw_enabled is False:
+            logger.debug(
+                "APM logs exports disabled. Skipping init of logs exporters"
+            )
+            return
+
+        # SolarWindsDistro._configure does setdefault before this is called
+        environ_exporter = os.environ.get(
+            OTEL_LOGS_EXPORTER,
+        )
+        if not environ_exporter:
+            logger.debug(
+                "No OTEL_LOGS_EXPORTER set, skipping init of logs exporters"
+            )
+            return
+
+        environ_exporter_names = environ_exporter.split(",")
+
+        # Use configured Resource attributes then merge with
+        # custom service.name and sw.trace_span_mode
+        resource = trace.get_tracer_provider().get_tracer(__name__).resource
+        sw_resource = Resource.create(
+            {
+                "sw.apm.version": __version__,
+                "sw.data.module": "apm",
+                "service.name": apm_config.service_name,
+            }
+        ).merge(resource)
+
+        logger_provider = LoggerProvider(
+            resource=sw_resource,
+        )
+        set_logger_provider(logger_provider)
+
+        for exporter_name in environ_exporter_names:
+            exporter = None
+            try:
+                exporter = next(
+                    iter_entry_points(
+                        "opentelemetry_logs_exporter", exporter_name
+                    )
+                ).load()()
+            except Exception as ex:
+                logger.exception("A exception was raised: %s", ex)
+                logger.exception(
+                    "Failed to load configured OTEL_LOGS_EXPORTER. "
+                    "Please reinstall or contact %s.",
+                    INTL_SWO_SUPPORT_EMAIL,
+                )
+                raise
+
+            if apm_config.is_lambda:
+                logger.debug(
+                    "Setting logs with SimpleLogRecordProcessor using %s",
+                    exporter_name,
+                )
+                logs_processor = SimpleLogRecordProcessor(exporter)
```

Comment:
I'm not super familiar with how Lambda instrumentation works for Python but in JS we still use batching processors and just flush at the end of the function invocation, is that something that would be possible here ?

Reply:
We did try that in Python previously, since the Batch processors also have force flush functions. I think we switched to the Simple processors across the board in part because of empty metrics messages that sometimes happened: https://github.com/solarwinds/apm-python/pull/312.

This might have been fixed in the upstream since APM Python finally upgraded, via https://github.com/open-telemetry/opentelemetry-python/releases/tag/v1.21.0 So I could try switching to Batch across the board again to simplify configuration logic! Is that ok in a different ticket?

Reply:
I created a new ticket! Will try switching back to Batch later.